components/style/context.rs

```diff
@@ -184,20 +184,9 @@ lazy_static! {
     };
 }
 
-#[cfg(feature = "servo")]
-fn shall_stat_style_sharing() -> bool {
-    opts::get().style_sharing_stats
-}
-
-#[cfg(not(feature = "servo"))]
-fn shall_stat_style_sharing() -> bool {
-    *DUMP_STYLE_STATISTICS
-}
-
 impl TraversalStatistics {
     /// Returns whether statistics dumping is enabled.
-    pub fn should_dump() -> bool {
-        shall_stat_style_sharing()
+        *DUMP_STYLE_STATISTICS || config::style_sharing_stats_enabled()
```

Comment:
This looks wrong. Currently servo returns `opts::get().style_sharing_stats` and gecko returns `*DUMP_STYLE_STATISTICS`. We need to preserve that behavior. You should remove the `*DUMP_STYLE_STATISTICS ||` from here and return `*DUMP_STYLE_STATISTICS` in gecko's style_sharing_stats_enabled function.